### ISSUE TITLE
Replace thread-per-dispatch with bounded completion architecture

### ISSUE THREAD

**Issue by sunway513** [sunway513/rocm-trace-lite#2]
### Problem
Every kernel dispatch spawns a detached thread:
```cpp
std::thread(signal_completion_handler, dd).detach();
```

For workloads with 100K+ dispatches (inference servers, fused microkernels, graph execution):
- Huge thread creation/destruction overhead
- Scheduler thrash, memory pressure from thread stacks
- Unbounded resource growth
- **Shutdown races**: detached workers still touching global state after `OnUnload()`/`close()`

### Impact
This is not suboptimal — it is a **likely failure mode** at production scale.

### Options (ordered by preference)
1. **`hsa_amd_signal_async_handler`** — HSA native async callback, zero thread overhead. Need to verify reliability across ROCm versions.
2. **Single poller thread** — One thread waiting on a queue of profiling signals. Simple, bounded.
3. **Bounded worker pool** — Fixed N threads, dispatch data queued. More complex but proven pattern (original rtg_tracer uses `ctpl::thread_pool`).

### Acceptance criteria
- No detached threads
- Bounded resource usage under 1M dispatches
- Clean shutdown: all workers joined before DB close
- No data loss at exit

### Files
- `src/hsa_intercept.cpp`: `queue_intercept_cb()`, `signal_completion_handler()`
